lib/foods.js

```diff
@@ -1,65 +1,75 @@
 $('document').ready(function() {
+  for (var i = 0; i < localStorage.length; i++){
+    $("tbody").append(localStorage.getItem(localStorage.key(i)));
+    add_delete_functionality();
+  }
+
   $("#add-food").on('click', function() {
     clear_errors();
 
     var name = $('#name-field input').val();
     var calories = $('#calories-field input').val();
-    
-    
+
+
     check_for_name(name)
     check_for_calories(calories)
     if (name == "" || calories == "") {
       return 1;
     }
 
     var foodRow = `<tr class='food-row'>
-                      <td class='food-name'>` + name + `</td>
-                      <td class='food-calories'>` + calories + `</td>
-                      <td class='food-delete'><button>-</button></td>
-                    </tr>`
+    <td class='food-name'>` + name + `</td>
+    <td class='food-calories'>` + calories + `</td>
+    <td class='food-delete'><button>-</button></td>
+    </tr>`
+    localStorage.setItem(name, foodRow);
     $("tbody").prepend(foodRow);
     add_delete_functionality();
     clear_text();
-  })
-  $("#food-filter input").on('keyup', function(){
-    var filter = $("#food-filter input").val().toLowerCase();
-    var foods = $("tbody").children();
-    for (var i = 0; i< foods.length; i++) {
-      var food = $(foods[i]).find('.food-name').text().toLowerCase();
-      if (!food.contains(filter)) {
-        $(foods[i]).slideUp();
-      }
-      else  {
-        $(foods[i]).slideDown();
+
+
+
+
+    $("#food-filter input").on('keyup', function(){
+      var filter = $("#food-filter input").val().toLowerCase();
+      var foods = $("tbody").children();
+      for (var i = 0; i< foods.length; i++) {
+        var food = $(foods[i]).find('.food-name').text().toLowerCase();
+        if (!food.contains(filter)) {
+          $(foods[i]).slideUp();
+        }
+        else  {
+          $(foods[i]).slideDown();
+        }
       }
-    }
+    })
   })
-})
 
-function check_for_name(name) {
-  if (name == ""){
+  function check_for_name(name) {
```

Comment:
Do these need to be wrapped in the $(document).ready()? It looks like they weren't before...